packages/frameworks-sveltekit/src/lib/webauthn.ts

```diff
@@ -0,0 +1,118 @@
+import { base } from "$app/paths"
+import { startAuthentication, startRegistration } from "@simplewebauthn/browser"
+
+import type {
+  BuiltInProviderType,
+  RedirectableProviderType,
+} from "@auth/core/providers"
+import type { WebAuthnOptionsResponseBody } from "@auth/core/types"
+import type { SignInOptions, SignInAuthorizationParams } from "./client.js"
+import type { LiteralUnion } from "./types.js"
+
+/**
+ * Fetch webauthn options from server and prompt user for authentication or registration.
+ * Returns either the completed WebAuthn response or an error request.
+ *
+ * @param providerId provider ID
+ * @param options SignInOptions
+ * @returns WebAuthn response or error
+ */
+async function webAuthnOptions(providerId: string, options?: SignInOptions) {
+  const baseUrl = `${base}/auth/`
+
+  // @ts-expect-error
+  const params = new URLSearchParams(options)
+
+  const optionsResp = await fetch(
+    `${baseUrl}/webauthn-options/${providerId}?${params}`
```

Comment:
Is there supposed to be a hook here? This URL is returning HTML, not JSON.

![CleanShot 2024-05-17 at 09 54 56@2x](https://github.com/nextauthjs/next-auth/assets/5138384/a98e0bd0-b849-4ce4-9b74-ac190c8674c3)

Reply:
Looks like there is an error in the console 
> InvalidProvider: Provider must be WebAuthn. Read more at https://errors.authjs.dev#invalidprovider